core/whistle-1.0.0/src/wh_util.erl

```diff
@@ -420,6 +420,37 @@ is_account_expired(Account) ->
 %%--------------------------------------------------------------------
 %% @public
 %% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_disable_account(ne_binary()) -> 'ok'.
+maybe_disable_account(Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:open_cache_doc(AccountDb, AccountId) of
+        {'ok', JObj} ->
+            case kz_account:is_enabled(JObj) of
+                'false' -> 'ok';
+                'true' -> disable_account(AccountId)
+            end;
+        {'error', _R} -> disable_account(AccountId)
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec disable_account(ne_binary()) -> 'ok'.
+disable_account(Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    case crossbar_maintenance:disable_account(AccountId) of
```

Comment:
This will need to be moved from `crossbar_maintenance` and `crossbar_maintenance` should call into the module that gets the code.